cypress/utils/dashboards/query_enhancement/commands.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+Cypress.Commands.add('waitForLoaderNewHeader', () => {
+  const opts = { log: false };
+
+  Cypress.log({
+    name: 'waitForPageLoad',
+    displayName: 'wait',
+    message: 'page load',
+  });
+  cy.wait(Cypress.env('WAIT_FOR_LOADER_BUFFER_MS'));
+  cy.getElementByTestId('recentItemsSectionButton', opts);
+});
+
+Cypress.Commands.add('setSingleLineQueryEditor', (value, submit = true) => {
+  const opts = { log: false };
+
+  Cypress.log({
+    name: 'setSingleLineQueryEditor',
+    displayName: 'set query',
+    message: value,
+  });
+
+  cy.getElementByTestId('osdQueryEditor__singleLine', opts).type(value, opts);
+
+  if (submit) {
+    cy.updateTopNav(opts);
+  }
+});
+
+Cypress.Commands.add('setQueryLanguage', (value) => {
+  Cypress.log({
+    name: 'setQueryLanguage',
+    displayName: 'set language',
+    message: value,
+  });
+
+  cy.getElementByTestId(`queryEditorLanguageSelector`).click();
+  cy.get(`[class~="languageSelector__menuItem"]`).contains(value).click({
```

Comment:
we could either add a test id which might be easier or access this class from the parent that has a test id of queryeditor langauge selector to ensure it doesn't grab the wrong language selector